Kirill_Samsonov/2/lib/spec/formatter_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require_relative '../formatter'
+
+describe Formatter do
+  subject { Class.new.extend Formatter }
+
+  let(:result) do
+    { domain: 'domain.com', status: 200, latency: 1500 }
+  end
+
+  it 'format valid domain result' do
+    result_string = 'domain.com - 200 (1500ms)'
+    expect(subject.format_domain_result(result)).to eq result_string
+  end
+
+  it 'format error result' do
+    result[:error] = true
+    result[:message] = '(Some error has happened)'
```

Comment:
It can be refactored like this:
```ruby
  let(:result) do
    { domain: 'domain.com', status: 200, latency: 1500, error: error, message: message }
  end
  
  let(:error) { nil }
  let(:message) { nil }
  
  ....
  
  
  it 'format error result' do
    let(:error) { true }
    let(:message) { '(Some error has happened)' }

```
So you will be able to implement one subject for multiple tests in the context and use just `expect(subject).to eq ...` or `is_expected.to eq ...`